solvers/ceres_solver.hpp

```diff
@@ -75,7 +78,7 @@ class CeresSolver : public karto::ScanSolver
   boost::mutex nodes_mutex_;
 
   // ros
-  rclcpp::Node::SharedPtr node_;
+  rclcpp::node_interfaces::NodeLoggingInterface::SharedPtr logging_interface_;
```

Comment:
You could just store the logger object instead, that seems easier. `rclcpp::Logger logger_; logger_ = node->get_logger()` and use `logger_` in the calls for logging.

Ditto across PR

Reply:
https://github.com/SteveMacenski/slam_toolbox/pull/561/commits/2ea6b342054962171e9278caef9ff3d96b6b8cc4

Updated to use logger objects. However `rclcpp::Logger` has a private constructor, so it is generated using a temporary name (`CeresSolver`) inside the header.